### PR TITLE
[Merged by Bors] - Implement `AggregateError`

### PR DESCRIPTION
**Pull request by HalidOdat**
It changes the following:
- Implement `AggregateError` global object

### CODE REVIEW

**Comment by HalidOdat** [comment]
Blocked until #1889 , gets merged

**Review by raskad — APPROVED** [review]

**Review by jedel1043 — APPROVED** [review]

**Pull request closed without merging**
